File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int len=s.size();
        int i=0,j=0,ans=0;
        bool vis[256]={false};
        while(j<len){
            if(vis[s[j]]){
                while(s[i]!=s[j]){
                    vis[s[i]]=false;
                    i++;
                }
                if(s[i]==s[j]){
                    vis[s[i]]=false;
                    i++;
                }
            }
            vis[s[j]]=true;
            ans=max(j-i+1,ans);
            j++;
        }
        ans=max(j-i,ans);
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        length = len(s)
        i = 0
        j = 0
        ans = 0
        vis = [False] * 256

        while j < length:
            if vis[ord(s[j])]:
                while s[i] != s[j]:
                    vis[ord(s[i])] = False
                    i += 1
                if s[i] == s[j]:
                    vis[ord(s[i])] = False
                    i += 1
            vis[ord(s[j])] = True
            ans = max(j - i + 1, ans)
            j += 1

        ans = max(j - i, ans)
        return ans
        